File: IT/1/Stringhe/Java/TestNumber.java
/* file TestNumber.java */

public class TestNumber {
	public static void main(String [] args) {
		String s = System.console().readLine("Digita un numero intero o in virgola mobile: ");

		/* controlla con due regex se "s" sia un numero intero/decimale */

		if (s.matches("^[0-9]+$")) {
			System.out.println(s + " è un numero intero.");
		}
		else if (s.matches("^[0-9]+\\.[0-9]{1,}$")) {
			System.out.println(s + " è un numero in virgola mobile.");
		}
		else {
			System.out.println("Attenzione! \"" + s + "\" non è un numero!");
		}
	}
}

File: IT/1/Stringhe/Python/test_number.py
# file test_number.py
# -*- coding: utf-8 -*-

from __future__ import print_function, unicode_literals
from re import match

# rinomina 'raw_input()' in 'input()' per compatibilità con Python 3
if hasattr(__builtins__, 'raw_input'): input = raw_input

s = input("Digita un numero intero o in virgola mobile: ")

# controlla con due regex se "s" sia un numero intero/decimale

if match(r"^[0-9]+$", s):
	print("{} è un numero intero.".format(s))

elif match(r"^[0-9]+\.[0-9]{1,}$", s):
	print("{} è un numero in virgola mobile.".format(s))

else:
	print("Attenzione! \"{}\" non è un numero!".format(s))

